ci: format checks

antoncoding opened the pull request:
# Summary
* added CI script to check code format
* added a cargo module that auto format term.toml, and have a `--check` flag that can be used for CI to check if all terms are sorted
* resolve #1 

### Run check and panic if not sorted:

```
cargo run -- --check
```

ChihChengLiang commented on check_terms/src/main.rs:
```diff
@@ -0,0 +1,55 @@
+use toml;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, Clone, PartialEq, Eq, Serialize)]
+struct Terms {
+    term: String,
+    translation: String,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct Root {
+    terms: Vec<Terms>,
+}
```
Sorry I forgot to tell you that we have the serialization/deserialization structs, parsing and dumping APIs, and a simple formattingg script for users defined in `src/`. See 

https://github.com/EtherTW/Terms/blob/4baa03f12f6dbd17fde54db7b82703530be57dc4/src/main.rs#L6-L35

and
 https://github.com/EtherTW/Terms/blob/4baa03f12f6dbd17fde54db7b82703530be57dc4/src/main.rs#L69-L70

ChihChengLiang commented on terms.toml:
```diff
@@ -2,6 +2,10 @@
 translation = "全節點"
 
+[[terms]]
+term = "Impermenant Loss"
```
Google suggests the following spelling 

```suggestion
term = "Impermanent Loss"
```

antoncoding replied:
ahhhhh good catch 🤣

ChihChengLiang commented on check_terms/Cargo.toml:
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "check_terms"
```
I see we created a new crate, "check_terms" here. 
We already have a root crate "terms". Instead of creating a new "check_terms" crate, can we implement the new features in the root crate "terms"?

The root module is defined here.
https://github.com/EtherTW/Terms/blob/4baa03f12f6dbd17fde54db7b82703530be57dc4/Cargo.toml#L1-L10

ChihChengLiang commented on check_terms/src/main.rs:
```diff
@@ -0,0 +1,55 @@
+fn main() {
+    let args: Vec<String> = env::args().collect();
+
+    let check_only = args.len() > 1 && args[1] == "--check";
```
Nice adding this `check` API.

ChihChengLiang commented on .github/workflows/format.yaml:
Great work adding the CI check

ChihChengLiang reviewed (commented):
Thanks for adding the CI check and the formatting check. I added some feedback.

I noticed some duplicated logic in the `check_terms/src/main.rs`. Can we move them to `src/main.rs`?
https://github.com/EtherTW/Terms/blob/4baa03f12f6dbd17fde54db7b82703530be57dc4/src/main.rs#L66-L74

If it's too much work, we can merge this pull request and fix in the next one.

antoncoding commented:
> Thanks for adding the CI check and the formatting check. I added some feedback.
> 
> I noticed some duplicated logic in the `check_terms/src/main.rs`. Can we move them to `src/main.rs`?


Totally reasonable. I was paying attention ti the root crate. 
Just added a `--check` flag into `main.rs`, and now we can just run `cargo run -- --check` on the root directory instead of switching between modules

ChihChengLiang reviewed (approved):
LGTM! Make perfect sense now.
